docs/src/modules/java/pages/quickstart/cr-value-entity-kafka-java.adoc

```diff
@@ -0,0 +1,154 @@
+
+== Define an Action
+
+The `customer_action.proto` will contain the definition of this action.
+
+. In your project, create one directories for your protobuf file, `src/main/proto/customer/action`.
+[.tabset]
+Linux or macOS::
++
+--
+[source,command line]
+----
+mkdir -p ./src/main/proto/customer/action
+----
+--
+Windows 10+::
++
+--
+[source,command line]
+----
+mkdir src/main/proto/customer/action
+----
+--
+
+. Create a `customer_action.proto` file and save it in the `src/main/proto/customer/action` directory.
+
+. Add declarations for:
++
+* The protobuf syntax version, `proto3`.
+* The package name, `customer.action`.
+* The required Java outer classname, `CustomerAction`. Messages defined in this file will be generated as inner classes.
+* Import `customer/domain/customer_domain.proto` and Akka Serverless `akkaserverless/annotations.proto`.
++
+[source,proto,indent=0]
+.src/main/proto/customer/action/customer_action.proto
+----
+include::example$java-customer-registry-kafka-quickstart/src/main/proto/customer/action/customer_action.proto[tag=declarations]
+----
+
+. Add the service definition. The service definition is annotated with `akkaserverless.codegen` indicating we want to generate an Action for this service.
+. Add declarations for:
++
+* Listening to `customer.domain.CustomerState` events from the value entity `customer` by using the option `eventing.in`.
+* Publishing to the Kafka topic `customer_upserts` by using the option `eventing.out`.
++
+[source,proto,indent=0]
+.src/main/proto/customer/action/customer_action.proto
+----
+include::example$java-customer-registry-kafka-quickstart/src/main/proto/customer/action/customer_action.proto[tag=service]
+----
+
+. Run `mvn compile` from the project root directory to generate source classes in which you add business logic.
++
+----
+mvn compile
+----
+
+== Create an Action
+
+Actions are stateless functions that can be triggered in multiple ways. In this case the action is triggered by each event `customer.domain.CustomerState` received by the Value Entity `customer.domain.Customer`. 
+
+. If it's not open already, open `src/main/java/customer/action/CustomerStateSubscriptionAction.java` for editing.
+
+. Modify the `onUpsertState` method by adding the logic to handle the action. The complete method should include the following:
++
+[source, java]
+.src/main/java/customer/action/CustomerStateSubscriptionAction.java
+----
+include::example$java-customer-registry-kafka-quickstart/src/main/java/customer/action/CustomerStateSubscriptionAction.java[tag=upsert]
+----
++
+* The incoming message contains the request data from your client and this action just passes it along to the xref:java:actions-publishing-subscribing.adoc[Pub/Sub] mechanism of choice. In this example Kafka.
+
+== Package and deploy your service
+
```

Comment:
Shouldn't there be some instructions (or link to other documentation) about how to setup Kafka (Confluent Cloud) and create the topic.

Reply:
I added that on the `Before you begin` section. Do you mean something else?

Reply:
Well, on that section above I didn't mention the name of the topic. I'll do so.

Reply:
ok, that's fine. I missed that.